### PR TITLE
HAAR-352 add additional address existing prison

### PR DESCRIPTION
**Pull request by AlTurner-MOJ**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/resource/PrisonAddressMaintenanceResource.kt [hunk 1]
```diff
@@ -89,6 +91,64 @@ class PrisonAddressMaintenanceResource(
     )
     return updatedAddress
   }
+
+  @PreAuthorize("hasRole('ROLE_MAINTAIN_REF_DATA') and hasAuthority('SCOPE_write')")
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/resource/PrisonAddressMaintenanceResource.kt:95]: I am wondering if all these should have the REGISTER_MAINTAINER role.  Will check with rest of team

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/service/PrisonAddressService.kt [hunk 2]
```diff
@@ -32,21 +34,51 @@ class PrisonAddressService(
...
+          addressLine2 = addressLine2,
+          town = town,
+          county = county,
+          postcode = postcode,
+          country = country,
+          prison = prison
+        )
+      )
+
+      recordPrisonAddressEditEvent("prison-register-add-address", prisonId, address.id?.toString(), additionalAddress)
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/service/PrisonAddressService.kt:60]: Not a deal breaker - but would probably have consistency in event name i.e. have both as prison-register-address-add/update or prison-register-add/update-address

**Review by karenmillermoj — APPROVED** [review]
lgtm
A few minor suggestions but happy for you to merge and address the comments separately

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/resource/PrisonAddressMaintenanceResource.kt (hunk 1 above)
Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/resource/PrisonAddressMaintenanceResource.kt:95]: Probably best as a separate ticket, depending on team view

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/service/PrisonAddressService.kt (hunk 2 above)
Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/service/PrisonAddressService.kt:60]: Hi Karen - agreed - updated for better consistency

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/resource/PrisonAddressMaintenanceResourceIntTest.kt [hunk 3]
```diff
@@ -182,6 +186,160 @@ class PrisonAddressMaintenanceResourceIntTest(@Autowired private val objectMappe
...
+          )
+        )
+        .exchange()
+        .expectStatus().isBadRequest
+      verifyNoInteractions(auditService, telemetryClient, prisonRepository, addressRepository)
+    }
+
+    @Test
+    fun `add a new address to an existing prison`() {
+      val prison = Prison("MDI", "A Prison", active = true)
```
[the remark this answers is not in the document]
Comment (AlTurner-MOJ) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/resource/PrisonAddressMaintenanceResourceIntTest.kt:263]: Hi Karen - I have tested this manually so know it works. The logic covered by this test is the same regardless of whether existing addresses are present or not

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**